internal/datanode/flush_manager.go

```diff
@@ -794,6 +798,13 @@ func flushNotifyFunc(dsService *dataSyncService, opts ...retry.Option) notifyMet
 				return fmt.Errorf(err.Error())
 			}
 
+			// Segment not found during stale segment flush. Segment might get compacted already.
+			// Return a specific error message and stop retry.
+			if !pack.flushed && rsp.GetErrorCode() == commonpb.ErrorCode_SegmentNotFound {
+				log.Error("stale segment not found, could be compacted",
+					zap.Int64("segment ID", pack.segmentID))
+				return retry.Unrecoverable(errors.New(staleSegmentNotFound))
+			}
```

Comment:
We could just return nil and print warning message instead of error string message check outsize